modules/unsupported/mbtiles/src/main/java/org/geotools/mbtiles/MBTilesFeatureSource.java

```diff
@@ -68,7 +61,10 @@ public MBTilesFeatureSource(
 
     @Override
     protected void addHints(Set<Hints.Key> hints) {
+
         hints.add(Hints.GEOMETRY_SIMPLIFICATION);
+        hints.add(Hints.GEOMETRY_GENERALIZATION);
+        hints.add(Hints.REMOVE_BUFFER_PIXEL);
```

Comment:
Here one could advertise the existing Hints.GEOMETRY_CLIP, so that the StreamingRenderer instantiates the clipping collection only when needed.

Reply:
fixed